server/controllers/conversation.js

```diff
@@ -0,0 +1,60 @@
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+const Profile = require("../models/Profile");
+
+exports.createConversationViaProfile = asyncHandler(async (req, res, next) => {
+  const senderId = req.user.id;
+  const { profileId } = req.body;
+  if (!profileId) {
+    res.status(400);
+    throw new Error("Bad Request");
+  }
+  const receiver = await Profile.findById(profileId);
+  !receiver && res.status(400).json({ error });
+  const receiverId = receiver.userId;
+  const conversation = await Conversation.create({
+    members: [senderId, receiverId],
+  });
+  if (!conversation) {
+    res.status(500);
+    throw new Error("Something went wrong with the conversation");
+  }
+
+  res.status(201).json(conversation);
+});
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { receiverId } = req.body;
+  console.log(req.user);
+  const senderId = req.user.id;
+  !receiverId && res.status(400).json({ error });
+
+  const existingConvo = await Conversation.find({
+    participants: { $all: [senderId, receiverId] },
+  });
+
+  if (existingConvo[0]) {
+    return res.status(200).json(existingConvo[0]);
+  }
+  const conversation = await Conversation.create({
+    members: [senderId, receiverId],
+  });
+  if (!conversation) {
+    res.status(500);
+    throw new Error("Something went wrong with the conversation");
+  }
+
+  res.status(201).json(conversation);
+});
+
+exports.getConversations = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const conversations = await Conversation.find({
+    members: { $in: [userId] },
```

Comment:
Additionally, you can also populate the last message here?

Reply:
Done